vm/systemSmartContracts/auction_test.go

```diff
@@ -0,0 +1,474 @@
+
+	minStakeValue := big.NewInt(10000000)
+	totalSupply := big.NewInt(100000000000)
+	minStep := big.NewInt(100000)
+	unBoundPeriod := uint64(100000)
+	numNodes := uint32(5)
+
+	arguments := CreateVmContractCallInput()
+	auctionData := createABid(25000000, 2, 12500000)
+	auctionDataBytes, _ := json.Marshal(&auctionData)
+
+	key1 := []byte("Key1")
+	key2 := []byte("Key2")
+	rewardAddr := []byte("tralala2")
+	maxStakePerNoce := big.NewInt(500)
+
+	eei := &mock.SystemEIStub{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		if bytes.Equal(key, arguments.CallerAddr) {
+			return auctionDataBytes
+		}
+		return nil
+	}
+	eei.SetStorageCalled = func(key []byte, value []byte) {
+		if bytes.Equal(key, arguments.CallerAddr) {
+			var auctionData AuctionData
+			_ = json.Unmarshal(value, &auctionData)
+			assert.Equal(t, big.NewInt(26000000), auctionData.TotalStakeValue)
+			assert.True(t, bytes.Equal(auctionData.BlsPubKeys[2], key1))
+			assert.True(t, bytes.Equal(auctionData.BlsPubKeys[3], key2))
+			assert.True(t, bytes.Equal(auctionData.RewardAddress, rewardAddr))
+			assert.Equal(t, maxStakePerNoce, auctionData.MaxStakePerNode)
+		}
+	}
+
+	kg := &mock.KeyGenMock{}
+
+	stakingAuctionSC, _ := NewStakingAuctionSmartContract(minStakeValue, minStep, totalSupply, unBoundPeriod, numNodes, eei, kg)
+
+	arguments.Function = "stake"
+	arguments.CallValue = big.NewInt(1000000)
+	arguments.Arguments = [][]byte{big.NewInt(2).Bytes(), key1, key2, maxStakePerNoce.Bytes(), rewardAddr}
+
+	errCode := stakingAuctionSC.Execute(arguments)
+	assert.Equal(t, vmcommon.Ok, errCode)
+}
+
+func TestStakingAuctionSC_ExecuteStakeUnStakeOneBlsPubKey(t *testing.T) {
+	t.Parallel()
+
+	minStakeValue := big.NewInt(10000000)
+	totalSupply := big.NewInt(100000000000)
+	minStep := big.NewInt(100000)
+	unBoundPeriod := uint64(100000)
+	numNodes := uint32(5)
+
+	arguments := CreateVmContractCallInput()
+	auctionData := createABid(25000000, 2, 12500000)
+	auctionDataBytes, _ := json.Marshal(&auctionData)
+
+	stakedData := StakedData{
+		StartNonce:    0,
+		Staked:        true,
+		UnStakedNonce: 1,
+		UnStakedEpoch: 0,
+		RewardAddress: []byte("tralala1"),
+	}
+	stakedDataBytes, _ := json.Marshal(&stakedData)
+
+	eei := &mock.SystemEIStub{}
+	kg := &mock.KeyGenMock{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		if bytes.Equal(key, arguments.CallerAddr) {
+			return auctionDataBytes
+		}
+		if bytes.Equal(key, auctionData.BlsPubKeys[0]) {
+			return stakedDataBytes
+		}
+		return nil
+	}
+	eei.SetStorageCalled = func(key []byte, value []byte) {
+		var stakedData StakedData
+		_ = json.Unmarshal(value, &stakedData)
+
+		assert.Equal(t, false, stakedData.Staked)
+	}
+
+	stakingAuctionSC, _ := NewStakingAuctionSmartContract(minStakeValue, minStep, totalSupply, unBoundPeriod, numNodes, eei, kg)
+
+	arguments.Function = "unStake"
+	arguments.Arguments = [][]byte{auctionData.BlsPubKeys[0]}
+	errCode := stakingAuctionSC.Execute(arguments)
+	assert.Equal(t, vmcommon.Ok, errCode)
```

Comment:
Should not check only the returned code for unStake (for staking you did check values)

Reply:
done

Reply:
Partially done, maybe check also the value